File: ProjectEuler/001-050/S38.cs
using System.Text;

namespace ProjectEuler
{
    public class S38 : ISolution
    {
        public string GetAnswer()
        {
            int maxPandigitalNumber = 0;
            for (int i = 2; i < 10000; i++)
            {
                StringBuilder sb = new StringBuilder();

                int n = 1;
                while (sb.Length < 9)
                {
                    sb.Append(i * n);
                    n++;
                }

                if (sb.Length == 9 && Utils.IsPandigital(sb.ToString(), false))
                {
                    int tmp = int.Parse(sb.ToString());

                    if (tmp > maxPandigitalNumber)
                    {
                        maxPandigitalNumber = tmp;
                    }
                }
            }

            return maxPandigitalNumber.ToString();
        }
    }
}

File: src/001-050/S38.cc
#include "S38.h"
#include "../Utils.h"

using namespace std;

namespace ProjectEuler
{
std::string S38::GetAnswer()
{
	int maxPandigitalNumber = 0;
	for (int i = 2; i < 10000; i++)
	{
		std::string sb = "";

		int n = 1;
		while (sb.size() < 9)
		{
			sb += std::to_string(i * n);
			n++;
		}

		if (sb.size() == 9 && Utils::IsPandigital(sb, false))
		{
			int tmp = std::stoi(sb);

			if (tmp > maxPandigitalNumber)
			{
				maxPandigitalNumber = tmp;
			}
		}
	}
	return std::to_string(maxPandigitalNumber);
}

} // namespace ProjectEuler
